src/Http/ControllerFactory.php

```diff
@@ -65,16 +91,7 @@ public function create(ServerRequest $request, Response $response)
             $this->missingController($request);
         }
 
-        $className = App::className($pluginPath . $controller, $namespace, 'Controller');
-        if (!$className) {
-            $this->missingController($request);
-        }
-        $reflection = new ReflectionClass($className);
-        if ($reflection->isAbstract() || $reflection->isInterface()) {
-            $this->missingController($request);
-        }
-
-        return $reflection->newInstance($request, $response, $controller);
+        return App::className($pluginPath . $controller, $namespace, 'Controller');
```

Comment:
return ... ?: null;

would be cleaner, returning more than null as alternative is a code smell usually.